### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by rolandschuetz** [jonnitto/Carbon.Link#2]
I love the package and use it a lot. In a recent project, I had a performance problem and debugged rendering a few hours. In the end, it came down to Carbon.Link.

I prepared a test case while rendering a simple link tag 1000x took ~7.5 sec on my notebook in dev context, the Carbon.LInk version took ~30.5 sec. That difference is massive.

Not yet sure how to solve it, but I found it worth documenting it here.

[Here's the test case](https://gist.github.com/rolandschuetz/d9a5985b414e6b0b6fbb09c53103a012)

**Comment by rolandschuetz** [comment]
@jonnitto Why do you convert a node to an identifier ([here](https://github.com/CarbonPackages/Carbon.Link/blob/master/Resources/Private/Fusion/Link/Link.fusion#L3)) to then convert it to a not with [${Neos.Link.convertUriToObject(link, node)}](https://github.com/CarbonPackages/Carbon.Link/blob/master/Resources/Private/Fusion/Link/URI.fusion#L19)?

And why do you [handle the section logic yoursel](https://github.com/CarbonPackages/Carbon.Link/blob/master/Resources/Private/Fusion/Link/URI.fusion#L20-L21)f, instead of using the NodeUri?

**Comment by jonnitto** [comment]
Yeah, I know, I should refactor this. This will be a breaking one…

**Comment by rolandschuetz** [comment]
What was the original intention behind this?
